modules/sync/sync.php

```diff
@@ -201,27 +201,36 @@ public function pll_save_post( $post_id, $post, $translations ) {
 	public function sync_term_parent( $term_id, $tt_id, $taxonomy ) {
 		global $wpdb;
 
-		if ( is_taxonomy_hierarchical( $taxonomy ) && $this->model->is_translated_taxonomy( $taxonomy ) ) {
-			$term = get_term( $term_id );
+		$action = explode( '_', current_filter() )[0]; // Either "created" or "edited".
+		if ( ! is_taxonomy_hierarchical( $taxonomy ) || ! $this->model->is_translated_taxonomy( $taxonomy ) ) {
+			return;
+		}
 
-			if ( $term instanceof WP_Term ) {
-				$translations = $this->model->term->get_translations( $term_id );
+		$term = get_term( $term_id );
+		if ( ! $term instanceof WP_Term ) {
+			return;
+		}
 
-				foreach ( $translations as $lang => $tr_id ) {
-					if ( $tr_id !== $term_id ) {
-						$tr_parent = $this->model->term->get_translation( $term->parent, $lang );
-						$tr_term   = get_term( (int) $tr_id, $taxonomy );
-
-						if ( $tr_term instanceof WP_Term && ! ( $term->parent && empty( $tr_parent ) ) ) {
-							$wpdb->update(
-								$wpdb->term_taxonomy,
-								array( 'parent' => $tr_parent ?: 0 ),
-								array( 'term_taxonomy_id' => $tr_term->term_taxonomy_id )
-							);
-
-							clean_term_cache( $tr_id, $taxonomy ); // OK since WP 3.9.
-						}
-					}
+		$translations = $this->model->term->get_translations( $term_id );
+
+		foreach ( $translations as $lang => $tr_id ) {
+			if ( $tr_id !== $term_id ) {
+				$tr_parent = $this->model->term->get_translation( $term->parent, $lang );
+				$tr_term   = get_term( (int) $tr_id, $taxonomy );
+
+				if ( 'created' === $action && 0 === $tr_parent ) {
```

Comment:
I would inline `$action` here as it's used only once. Taking the idea of Viktor, this leads to:
```suggestion
				if ( str_starts_with(current_filter(), 'created_') && 0 === $tr_parent ) {
```